python/lib/sift_py/data_import/csv.py

```diff
@@ -0,0 +1,198 @@
+                "Authorization": f"Bearer {self._apikey}",
+                "Content-Encoding": "application/octet-stream",
+            },
+            data=json.dumps({"csv_config": csv_config.to_dict()}),
+        )
+
+        if response.status_code != 200:
+            raise Exception(
+                f"Config file upload request failed with status code {response.status_code}. {response.text}"
+            )
+
+        try:
+            upload_info = response.json()
+        except (json.decoder.JSONDecodeError, KeyError) as e:
+            raise Exception(f"Invalid response: {e}")
+
+        try:
+            upload_url: str = upload_info["uploadUrl"]
+            data_import_id: str = upload_info["dataImportId"]
+        except KeyError as e:
+            raise Exception(f"Response missing required keys: {e}")
+
+        with open(path, "rb") as f:
+            headers = {
+                "Authorization": f"Bearer {self._apikey}",
+                "Content-Encoding": content_encoding,
+            }
+
+            response = requests.post(
+                url=upload_url,
+                headers=headers,
+                data=f,
+            )
+
+            if response.status_code != 200:
+                raise Exception(
+                    f"Data file upload request failed with status code {response.status_code}. {response.text}"
+                )
+
+            return DataImportStatus(self._rest_conf, data_import_id)
+
+    def upload_from_url(
+        self,
+        url: str,
+        csv_config: CsvConfig,
+    ) -> DataImportStatus:
+        parsed_url = urlparse(url)
+        if parsed_url.scheme not in ["s3", "http", "https"]:
+            raise Exception(
+                f"Invalid URL scheme: '{parsed_url.scheme}'. Only S3 and HTTP(S) URLs are supported."
+            )
+
+        headers = {"Authorization": f"Bearer {self._apikey}"}
+
+        response = requests.post(
+            url=self._url_uri,
+            headers=headers,
+            data=json.dumps(
+                (
+                    {
+                        "url": url,
+                        "csv_config": csv_config.to_dict(),
+                    }
+                )
+            ),
+        )
+
+        if response.status_code != 200:
+            raise Exception(
+                f"URL upload request failed with status code {response.status_code}. {response.text}"
+            )
+
+        try:
+            upload_info = response.json()
+        except (json.decoder.JSONDecodeError, KeyError) as e:
+            raise Exception(f"Invalid response: {e}")
+
+        try:
+            data_import_id: str = upload_info["dataImportId"]
+        except KeyError as e:
+            raise Exception(f"Response missing required keys: {e}")
+
+        return DataImportStatus(self._rest_conf, data_import_id)
+
+    def simple_upload(
+        self,
+        asset_name: str,
+        path: Union[str, Path],
+        run_name: Optional[str] = None,
```

Comment:
For args that default to `None` we should make them the last args in the function signature so they can omit them when calling, otherwise they'll have to explicitly pass in `None`.